fetch-api/js/app.js

```diff
@@ -0,0 +1,70 @@
+const select = document.getElementById('breeds');
+const card = document.querySelector('.card');
+
+select.addEventListener('change', fetchBreedImage);
+card.addEventListener('click', fetchBreedImage);
+document.querySelector('form').addEventListener('submit', postInputFieldData);
+
+Promise.all([
+    fetchData('https://dog.ceo/api/breeds/image/random'),
+    fetchData('https://dog.ceo/api/breeds/list')
+])
+    .then(data => {
+        generateImage(data[0].message);
+        generateBreedOptions(data[1].message);
+    });
+
+//Functions:
+function checkStatus(response) {
+    if (response.ok) {
+        return Promise.resolve(response);
+    }
+    return Promise.reject(new Error(response.statusText));
+}
+
+function generateImage(data) {
+    card.innerHTML = `
+        <img src='${data}' alt>
+        <p>Click to view images of ${select.value}s</p>
+        `;
+}
+
+function generateBreedOptions(data) {
+    select.innerHTML = data.map(item => `
+        <option value='${item}'>${item}</option>
+        `).join("");
+}
+
+function fetchBreedImage() {
+    const breed = select.value;
```

Comment:
I think I would rather see breed as an argument. That should make it more flexible.
Right now you have this concept of a breed HTML element via the select var for an id tag `tightly coupled` with the concept of getting a breed image from an external API.

Ideally I would like to see the API call totally by itself and a separate function where you act upon that data.

Lets talk about this if you need more explanation.

Reply:
Done.